### PR TITLE
Better keyboard support for distributor admin menu

### LINKED ISSUE #563: Distributor menu in admin bar - better keyboard support
## Is your enhancement related to a problem? Please describe.
- Can’t select individual connection with keyboard. Or un-select.
- Focus styles missing inside the “Distributor menu”.
- I’d show connections as checkboxes in the first place. Now they are div or span.

## Describe the solution you'd like
Re-think the interface with accessible markup and keyboard support.

Related issue is #557

### PR DESCRIPTION
**Pull request by samikeijonen**
Fixes #563

## Description of the Change

- Use `<button>` instead of `<div>`.
- Use Core button styles where possible for consistency and better focus styles.
- Use `disabled` attribute when post have been syndicated.
- This is not full refactoring but first step in better accessibility.

## Alternate Designs

None.

## Benefits

Better keyboard support when using correct semantics out of the box.

## Possible Drawbacks

Complex codebase, needs testing that nothing is broken.

## Verification Process

Distribute and push article using only keyboard. 

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/10up/distributor/blob/develop/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

## Applicable Issues

#563

### CODE REVIEW

**Review by dinhtungdu — CHANGES_REQUESTED** [review]
The code looks good @samikeijonen. There is a small issue that internal connections can't be selected using keyboard because they're still using `div`. Can you please update?

File: assets/css/push.css [hunk 1]
```diff
@@ -177,55 +175,43 @@
...
-	border-width: 1px;
-    border-style: solid;
-	box-shadow: 0 1px 0 #006799;
-	color: #fff;
-	text-decoration: none;
-	text-shadow: 0 -1px 1px #006799, 1px 0 1px #006799, 0 1px 1px #006799, -1px 0 1px #006799;
-	height: 30px;
-	line-height: 28px;
-	padding: 0 12px 2px;
-	border-radius: 3px;
```
Comment (dinhtungdu) [inline assets/css/push.css:191]: we should keep the `border-radius` and `text-shadow` because they're overridden by admin bar style.

**Comment by samikeijonen** [comment]
@dinhtungdu Unfortunately I have zero hours for updates at the moment.

**Comment by dinhtungdu** [comment]
@samikeijonen no problem, I will take over this.

**Review by dinhtungdu — DISMISSED** [review]

**Comment by dkotter** [comment]
@dinhtungdu Testing this locally, I'm seeing the connection list break if you have more than 4 connections:

Screen Shot 2020-05-15 at 11 28 51 AM

I believe once there are 5 or more connections, we constrain those to a box that you can scroll through, to keep the list down. Looks like something changed here that is preventing that from working properly, so the entire connection list is shown and there's no way to scroll through them all.

**Comment by dinhtungdu** [comment]
Thank @dkotter, you're right, the style to handle overflow was removed. I added it again, the list is now working correctly with more than 4 connections.

**Review by dkotter — APPROVED** [review]

**Comment by dkotter** [comment]
@dinhtungdu Looks good 👍

**Pull request merged**